consensus/leader.go

```diff
@@ -303,9 +303,15 @@ func (consensus *Consensus) onCommit(msg *msg_pb.Message) {
 			consensus.preCommitAndPropose(blockObj)
 		}
 
-		consensus.getLogger().Info().Msg("[OnCommit] Starting Grace Period")
 		go func(viewID uint64) {
-			time.Sleep(1000 * time.Millisecond)
+			waitTime := 1000 * time.Millisecond
+			maxWaitTime := time.Until(consensus.NextBlockDue)
+			if maxWaitTime > waitTime {
```

Comment:
From the variable name I read the condition is supposed to be `waitTime > maxWaitTime`?

Reply:
not really, we want the waitTime to be Max(1s, <rest of the block time>)